### PR TITLE
Fix notebook bugs for branch r1.8.0

### PR DESCRIPTION
**Pull request by yidong72**
# What does this PR do ?

Fixed the biomegatron model NGC link error.
`get_lm_model` can get biomegatron model through model names.
Fix the biomegatron notebooks to directly load the biomegatron from NGC.
Fix the MegatronLM ckpt conversion error. 
Fix the typos in synthetic data generation notebook.

### CODE REVIEW

File: examples/nlp/language_modeling/megatron_lm_ckpt_to_nemo.py [hunk 1]
```diff
@@ -360,7 +360,7 @@ def convert(local_rank, rank, world_size, args):
     trainer = Trainer(devices=args.gpus_per_node, accelerator='gpu', num_nodes=num_nodes)
 
     # TODO: reach out to PTL For an API-safe local rank override
-    trainer.accelerator.training_type_plugin._local_rank = local_rank
+    # trainer.accelerator.training_type_plugin._local_rank = local_rank
```
Comment (ericharper) [inline examples/nlp/language_modeling/megatron_lm_ckpt_to_nemo.py:363]: Is this not needed anymore? Let's delete the comment.

File: nemo/collections/nlp/models/nlp_model.py [hunk 2]
```diff
@@ -76,20 +76,26 @@ def __init__(self, cfg: DictConfig, trainer: Trainer = None, no_lm_init=False):
                 config_dict = OmegaConf.to_container(cfg.language_model.config)
             if cfg.get('language_model.config_file'):
                 config_file = self.register_artifact('language_model.config_file', cfg.language_model.config_file)
-            self.bert_model = get_lm_model(
+            bert_model = get_lm_model(
```
Comment (ericharper) [inline nemo/collections/nlp/models/nlp_model.py:79]: Shouldn't this be under a conditional?

File: examples/nlp/language_modeling/megatron_lm_ckpt_to_nemo.py (hunk 1 above)
Comment (yidong72) [reply examples/nlp/language_modeling/megatron_lm_ckpt_to_nemo.py]: deleted.

**Review by borisfom — CHANGES_REQUESTED** [review]
This PR rolls back many of [@Ramanathan Arunachalam](https://nvidia.slack.com/team/U020UN35TGF)’s functional changes - probably because of bad merge. 
Please examine diffs and do 'git fetch origin; git checkout -r <file>' for files you did not mean to touch. Carefully inspect other files that only your changes are include in the diff.

**Comment by yidong72** [comment]
> This PR rolls back many of [@Ramanathan Arunachalam](https://nvidia.slack.com/team/U020UN35TGF)’s functional changes in main (#3919) - probably because of bad merge. for example: [examples/nlp/language_modeling/megatron_lm_ckpt_to_nemo.py](https://github.com/NVIDIA/NeMo/pull/3948/files#diff-c856b5e10569b76943b062527702532bb2f2034798422044086e315a61409ac4) [nemo/collections/nlp/models/nlp_model.py](https://github.com/NVIDIA/NeMo/pull/3948/files#diff-ab7ae31d24e5872426961bd8ef793bed63a18c9f4473bac372e00b1a07a9e36c)]
> 
> Please examine diffs and do 'git fetch origin; git checkout -r ' for files you did not mean to touch. Carefully inspect other files that only your changes are include in the diff.

[examples/nlp/language_modeling/megatron_lm_ckpt_to_nemo.py](https://github.com/NVIDIA/NeMo/pull/3948/files#diff-c856b5e10569b76943b062527702532bb2f2034798422044086e315a61409ac4) is correct. 
I changed the `nlp_model` and `lm_util` to be compatible with the [@Ramanathan Arunachalam's](https://nvidia.slack.com/team/U020UN35TGF) changes. A lot of the changes are made to allow MegatronBERT model load from NGC. 
I also added a Jenkins test to cover the biomegatron fine-tuning.
Please help to review it.

**Review by borisfom — APPROVED** [review]

**Review by ericharper — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**